src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -279,41 +277,127 @@ private Void showTrackingNotification() {
+                    addModeInfo(windowTitle, bibDatabaseContext);
+
+                    if (panel.isModified() && !isAutosaveEnabled) {
+                        addChangedInformation(toolTipText, fileName);
+                    }
 
-            setWindowTitle();
-            updateAllTabTitles();
+                    String uniquePath = uniquePathParts.get(i);
+                    if (!uniquePath.equals(fileName) && uniquePath.contains(File.separator)) {
+                        // remove filename
+                        uniquePath = uniquePath.substring(0, uniquePath.lastIndexOf(File.separator));
+                        tabTitle.append(" \u2013 ").append(uniquePath);
+                        windowTitle.append(" \u2013 ").append(uniquePath);
+                    }
+                } else {
+                    if (databaseLocation == DatabaseLocation.LOCAL) {
+                        tabTitle.append(Localization.lang("untitled"));
+                        windowTitle.append(Localization.lang("untitled"));
+                        if (bibDatabaseContext.getDatabase().hasEntries()) {
+                            // if the database is not empty and no file is assigned,
+                            // the database came from an import and has to be treated somehow
+                            // -> mark as changed
+                            tabTitle.append('*');
+                            windowTitle.append('*');
+                        }
+                    } else {
+                        addSharedDbInformation(tabTitle, bibDatabaseContext);
+                        addSharedDbInformation(toolTipText, bibDatabaseContext);
+                    }
+                    addModeInfo(toolTipText, bibDatabaseContext);
+                    addModeInfo(windowTitle, bibDatabaseContext);
+                    if (databaseLocation == DatabaseLocation.LOCAL && bibDatabaseContext.getDatabase().hasEntries()) {
+                        addChangedInformation(toolTipText, Localization.lang("untitled"));
+                    }
+                }
+
+                tabbedPane.getTabs().get(i).setText(tabTitle.toString());
+                tabbedPane.getTabs().get(i).setTooltip(new Tooltip(toolTipText.toString()));
+
+                if (i == selectedTabIndex) {
+                    windowTitle.append(" \u2013 ");
+                    windowTitle.append(FRAME_TITLE);
+                    mainStage.setTitle(windowTitle.toString());
+                }
+            }
         });
     }
 
-    /**
-     * Sets the title of the main window.
-     */
-    public void setWindowTitle() {
-        BasePanel panel = getCurrentBasePanel();
-
-        // no database open
-        if (panel == null) {
-            //setTitle(FRAME_TITLE);
-            return;
-        }
+    private void addChangedInformation(StringBuilder text, String fileName) {
+        text.append(". ");
+        text.append(Localization.lang("Library '%0' has changed.", fileName));
```

Comment:
As this is displayed only in the tooltip, it's always clear to which database it is referring.